src/ScriptEditor/ui/ScriptEditorRoot.tsx

```diff
@@ -628,8 +558,8 @@ export function Root(props: IProps): React.ReactElement {
     const server = GetServer(closingScript.hostname);
     if (server === null) throw new Error(`Server '${closingScript.hostname}' should not be null, but it is.`);
 
-    const serverScriptIndex = server.scripts.findIndex((script) => script.filename === closingScript.fileName);
-    if (serverScriptIndex === -1 || savedScriptCode !== server.scripts[serverScriptIndex as number].code) {
+    const oldScript = server.getScript(closingScript.fileName);
+    if (oldScript !== null && oldScript.code !== savedScriptCode) {
```

Comment:
is savedScriptCode still safe? looks like before it used all scripts not its open scripts only... though it doesnt make sense it would close a tab thats not open. (maybe it was trying to close ghost scripts?)